diff --git a/src/poetry/core/masonry/builders/builder.py b/src/poetry/core/masonry/builders/builder.py
index 0b450e3..24acc4b 100644
--- a/src/poetry/core/masonry/builders/builder.py
+++ b/src/poetry/core/masonry/builders/builder.py
@@ -1,6 +1,7 @@
 from __future__ import annotations
 
 import logging
+import os
 import re
 import sys
 import warnings
@@ -108,8 +109,11 @@ class Builder:
 
             # Checking VCS
             vcs = get_vcs(self._path)
-            vcs_ignored_files = set(vcs.get_ignored_files()) if vcs else set()
-
+            if vcs and "CONDA_BUILD" not in os.environ and \
+                os.environ.get("POETRY_NO_VCS_IGNORED_FILES", "0") != "1":
+                    vcs_ignored_files = set(vcs.get_ignored_files())
+            else:
+                vcs_ignored_files = set()
             explicitly_excluded = set()
             for excluded_glob in self._package.exclude:
                 for excluded in self._path.glob(str(excluded_glob)):
