diff -ruN poetry_core-2.0.0/src/poetry/core/masonry/builders/builder.py poetry_core-2.0.0-patch/src/poetry/core/masonry/builders/builder.py
--- poetry_core-2.0.0/src/poetry/core/masonry/builders/builder.py	2025-01-04 18:42:15.423561195 +0100
+++ poetry_core-2.0.0-patch/src/poetry/core/masonry/builders/builder.py	2025-01-04 18:45:53.568840605 +0100
@@ -1,6 +1,7 @@
 from __future__ import annotations
 
 import logging
+import os
 import sys
 import textwrap
 
@@ -88,8 +89,11 @@
 
             # Checking VCS
             vcs = get_vcs(self._path)
-            vcs_ignored_files = set(vcs.get_ignored_files()) if vcs else set()
-
+            if vcs and "CONDA_BUILD" not in os.environ and \
+                    os.environ.get("POETRY_NO_VCS_IGNORED_FILES", "0") != "1":
+                vcs_ignored_files = set(vcs.get_ignored_files())
+            else:
+                vcs_ignored_files = set()
             explicitly_excluded = set()
             for excluded_glob in self._package.exclude:
                 for excluded in self._path.glob(str(excluded_glob)):
